LPS-122885 Properly export delegate utility

kresimir-coko opened the pull request:
@bryceosterhaus I've taken liberty and made this addition to your #439 PR.

I've tested this by importing it in `UserCard.js` in our Clay Sample module to make sure it's getting properly imported in JS files, and also tested by trying to access it via `Liferay.Util.delegate` in the console.

The reason why I added the `export {delegate}` alongside the `export default delegate` is because that's the approach taken in other utilities, for example in [debounce.es.js](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/debounce/debounce.es.js#L39-L40)

jbalsas commented on modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js:
```diff
@@ -54,3 +54,4 @@ function delegate(element, eventName, selector, callback) {
 }
 
 export default delegate;
+export {delegate};
```
> The reason why I added the export {delegate} alongside the export default delegate is because that's the approach taken in other utilities, for example in debounce.es.js

See https://github.com/liferay-frontend/liferay-portal/pull/439#discussion_r511969174

> Do not export the same thing twice; I know you're just following the old convention that was used in some Liferay codebases, but we now consider it an anti-pattern. [The linter will prevent this in the future](https://github.com/liferay/liferay-frontend-projects/issues/4).

wincent replied:
Beware of cargo-culting: consistency is great and all, but it's worth taking the time to understand _why_ something is being done in a certain way. In this case, we concluded there's no good reason to do this any more (hence the lint, which I guess I have to prioritize because this keeps coming up).

kresimir-coko replied:
Makes sense, thanks guys!

kresimir-coko commented:
@wincent @jbalsas I've pushed the removal of the anti-pattern.

jbalsas commented:
ci:test:sf

jbalsas commented:
ci:test:relevant

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-22.liferay.com/job/test-portal-source-format/5846/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-22.liferay.com//userContent/jobs/test-portal-source-format/builds/5846/jenkins-report.html">here</a>.</b></p>

jbalsas commented:
Hey @kresimir-coko, I think I don't have permissions to push to your origin. Could you please check and also invite the other reviewers if they don't have either?

I was going to squash that last commit. Maybe it's just me, but it confuses me when we add and remove code in back to back commits in a same PR. Unless there's some really useful information you need to convey, just squash them in the future to avoid this weirdness.

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/02396f764ad152bde79f062c7cb1ca941a670234">02396f764ad152bde79f062c7cb1ca941a670234</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/kresimir-coko/liferay-portal/tree/LPS-122885">LPS-122885</a><br/>Branch GIT ID: <a href="https://github.com/kresimir-coko/liferay-portal/commit/598dbd37cfb69c991b0fb9fb6df1bbf4cecfb898">598dbd37cfb69c991b0fb9fb6df1bbf4cecfb898</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-22.liferay.com/job/test-portal-source-format/5846/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-22.liferay.com/userContent/jobs/test-portal-source-format/builds/5846/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<strong>Jenkins Build:</strong><a href="https://test-1-22.liferay.com/job/test-portal-source-format/5846/">test-portal-source-format#5846</a><br /><strong>Jenkins Report:</strong><a href="https://test-1-22.liferay.com/userContent/jobs/test-portal-source-format/builds/5846/jenkins-report.html">jenkins-report.html</a><br /><strong>Jenkins Suite:</strong> sf<br /><strong>Pull Request:</strong><a href="https://github.com/liferay-frontend/liferay-portal/pull/464">liferay-frontend#464</a><br /><strong>Spira Release:</strong><a href="https://liferay.spiraservice.net/16/Release/1302.aspx">/Liferay DXP 7.3/Pull Request/ci:test:sf</a><br /><strong>Spira Release Build:</strong><a href="https://liferay.spiraservice.net/16/Build/56073.aspx">master - kresimir-coko > liferay-frontend - PR#464 - 2020-10-30[11:01:34]</a><br /><strong>Spira Jenkins Build:</strong><a href="https://test-1-20.liferay.com/job/publish-spira-report/6519/">publish-spira-report#6519</a><br />

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest(master)/7466/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-22.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/7466/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 9 out of 9 jobs passed</h3><h3>:heavy_check_mark: ci:test:relevant - 22 out of 23 jobs passed in 1 hour 58 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/ad24aa801471833be3a2c21eae3976b5066ab88c">ad24aa801471833be3a2c21eae3976b5066ab88c</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal-ee/tree/master-private">master-private</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal-ee/commit/424ab1158c689b11bc409031cffca77c71b5b194">424ab1158c689b11bc409031cffca77c71b5b194</a></p><details><summary><strong>ci:test:stable - 9 out of 9 jobs PASSED</strong></summary><details><summary><strong>9 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340815/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/477127/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/407169/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/443287/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/449121/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/443288/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/407172/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/415879/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/367585/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable</a></li></ul></details></details><details><summary><strong>ci:test:relevant - 21 out of 23 jobs PASSED</strong></summary><div><h4>2 Failed Jobs:</h4><ul><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest(master)/7466/"><strike><strong>test-portal-acceptance-pullrequest(master)</strong></strike></a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/369838/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</strong></strike></a></li></ul></div><details><summary><strong>21 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/415880/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340816/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340815/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/415881/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/477127/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/407169/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/407171/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/443286/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/340817/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/443289/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/443287/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/407170/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/426024/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/449121/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/426025/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/443288/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/407172/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/426026/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/415879/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/477128/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/367585/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable</a></li></ul></details></details><h5>For more details click <a href="https://test-1-22.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/7466/jenkins-report.html">here</a>.</h5><hr/><h4>This pull contains no unique failures.</h4><hr/><details><summary><strong>Failures in common with <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">acceptance upstream results</a> at 02396f764ad152bde79f062c7cb1ca941a670234:</strong></summary><ol><li><div><h5><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/369838/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</a></h5><div><h6>Job Results:</h6><p>40 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/369838//consoleText">AXIS_VARIABLE=0,label_exp=!master #369838</a><ol><li><div><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/369838//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_ImageSelector_ViewImageEditorControls_">LocalFile.ImageSelector#ViewImageEditorControls</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-22/1604082183222/test-portal-acceptance-pullrequest(master)/7466/functional-tomcat90-mysql57-jdk8/0/0/LocalFile.ImageSelector_ViewImageEditorControls/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-22/1604082183222/test-portal-acceptance-pullrequest(master)/7466/functional-tomcat90-mysql57-jdk8/0/0/LocalFile.ImageSelector_ViewImageEditorControls/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/test-1-22/1604082183222/test-portal-acceptance-pullrequest(master)/7466/functional-tomcat90-mysql57-jdk8/0/0/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li></ol></div></li></ol></details></details></html>

liferay-continuous-integration commented:
<strong>Jenkins Build:</strong><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest(master)/7466/">test-portal-acceptance-pullrequest(master)#7466</a><br /><strong>Jenkins Report:</strong><a href="https://test-1-22.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/7466/jenkins-report.html">jenkins-report.html</a><br /><strong>Jenkins Suite:</strong> relevant<br /><strong>Pull Request:</strong><a href="https://github.com/liferay-frontend/liferay-portal/pull/464">liferay-frontend#464</a><br /><strong>Spira Release:</strong><a href="https://liferay.spiraservice.net/16/Release/1312.aspx">/Liferay DXP 7.3/Pull Request/ci:test:relevant</a><br /><strong>Spira Release Build:</strong><a href="https://liferay.spiraservice.net/16/Build/56157.aspx">master - kresimir-coko > liferay-frontend - PR#464 - 2020-10-30[11:23:03]</a><br /><strong>Spira Jenkins Build:</strong><a href="https://test-1-4.liferay.com/job/publish-spira-report/8471/">publish-spira-report#8471</a><br />

jbalsas commented on modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/index.es.js:
```diff
@@ -26,6 +26,10 @@ export {default as autoSize} from './liferay/autosize/autosize.es';
 
+// Delegate API
+
+export {delegate} from './liferay/delegate/delegate.es';
```
I think this shouldn't work now, since you're importing named but exporting default... 🤔

kresimir-coko replied:
That makes sense, but I tested it and it worked, maybe some caching was going on, even though I noticed changes were being made. Interesting, I'll check it out.

kresimir-coko commented:
@jbalsas I've squashed the commits and pushed. I've also added you to the list of contributors, not sure if that's the way to do it, I've never had issues with that in the past so first time having to do this.

![image](https://user-images.githubusercontent.com/24564752/97845335-c0f3e100-1cec-11eb-998c-d8413b73eca6.png)

bryceosterhaus commented on modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js:
```diff
@@ -28,6 +28,7 @@ import {
 	getComponentCache,
 	initComponentCache,
 } from './component.es';
+import delegate from './delegate/delegate.es';
```
I think we should avoid providing this as a global since this would be a new delivery for `delegate`. See https://liferay.slack.com/archives/G3JBR21HA/p1604343092181200?thread_ts=1604053827.155100&cid=G3JBR21HA for more convo on that

kresimir-coko replied:
When I remove `delegate` from this file I can't see `delegate` in the `Liferay.Util` object, I think that's the point but I'm not sure how I can make sure JSPs can access it in this case. I'm not 100% sure about how all this hooks up without the import and export in `global.es.js` @wincent got any thoughts?

jbalsas replied:
Hey @kresimir-coko, that's the point, it wasn't there before, no reason it should be there now.

Before, in all cases, you were using it like `require="metal-dom/src/...."` so now you should be able to do something like `require="frontend-js-web"` and then get the `delegate` method from there.

jbalsas replied:
This is technically also wasteful since we're downloading most of the code twice in `global.es.js` and through the `main` entry in `frontend-js-web`,  but that's something we'll need to figure out later at a lower level.

kresimir-coko replied:
Okay, that makes sense, the `require` part was missing for me, I don't remember seeing `require="frontend-js-web"` elsewhere, but I guess that's the goal, we want to make it the go-to way of importing our global `Liferay.Util` object.

bryceosterhaus commented on modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/index.es.js:
```diff
@@ -26,6 +26,10 @@ export {default as autoSize} from './liferay/autosize/autosize.es';
 
 export {cancelDebounce, debounce} from './liferay/debounce/debounce.es';
 
+// Delegate API
+
+export {default as delegate} from './liferay/delegate/delegate.es';
+
```
Thanks for catching this and updating this for me!

kresimir-coko commented:
@jbalsas @wincent  Sent an updated commit that works properly after testing it.

wincent commented:
ci:test:sf

wincent commented:
ci:test:relevant

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-23.liferay.com/job/test-portal-source-format/5036/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-23.liferay.com//userContent/jobs/test-portal-source-format/builds/5036/jenkins-report.html">here</a>.</b></p>

wincent reviewed (approved):
lg2m

Feel free to `ci:forward` as soon as CI is green (again) on this.

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 3 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/afb229df01a027b6f438bf1628290d9470d1e4d8">afb229df01a027b6f438bf1628290d9470d1e4d8</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/kresimir-coko/liferay-portal/tree/LPS-122885">LPS-122885</a><br/>Branch GIT ID: <a href="https://github.com/kresimir-coko/liferay-portal/commit/fd1cb346f5e33f4f20ec5077393700de42d295ca">fd1cb346f5e33f4f20ec5077393700de42d295ca</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-23.liferay.com/job/test-portal-source-format/5036/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-23.liferay.com/userContent/jobs/test-portal-source-format/builds/5036/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<strong>Jenkins Build:</strong><a href="https://test-1-23.liferay.com/job/test-portal-source-format/5036/">test-portal-source-format#5036</a><br /><strong>Jenkins Report:</strong><a href="https://test-1-23.liferay.com/userContent/jobs/test-portal-source-format/builds/5036/jenkins-report.html">jenkins-report.html</a><br /><strong>Jenkins Suite:</strong> sf<br /><strong>Pull Request:</strong><a href="https://github.com/liferay-frontend/liferay-portal/pull/464">liferay-frontend#464</a><br /><strong>Spira Release:</strong><a href="https://liferay.spiraservice.net/16/Release/1302.aspx">/Liferay DXP 7.3/Pull Request/ci:test:sf</a><br /><strong>Spira Release Build:</strong><a href="https://liferay.spiraservice.net/16/Build/56876.aspx">master - kresimir-coko > liferay-frontend - PR#464 - 2020-11-03[03:43:47]</a><br /><strong>Spira Jenkins Build:</strong><a href="https://test-1-16.liferay.com/job/publish-spira-report/10561/">publish-spira-report#10561</a><br />

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest(master)/6881/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-23.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6881/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 9 out of 9 jobs passed</h3><h3>:heavy_check_mark: ci:test:relevant - 23 out of 23 jobs passed in 2 hours 12 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/afb229df01a027b6f438bf1628290d9470d1e4d8">afb229df01a027b6f438bf1628290d9470d1e4d8</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal-ee/tree/master-private">master-private</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal-ee/commit/8d22a791300569a6c92e7b9152e6ae1c6bdc863a">8d22a791300569a6c92e7b9152e6ae1c6bdc863a</a></p><details><summary><strong>ci:test:stable - 9 out of 9 jobs PASSED</strong></summary><details><summary><strong>9 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/445750/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/407378/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/454743/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/445751/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/445755/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/454746/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/445754/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/454744/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/369973/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable</a></li></ul></details></details><details><summary><strong>ci:test:relevant - 23 out of 23 jobs PASSED</strong></summary><details><summary><strong>23 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest(master)/6881/">test-portal-acceptance-pullrequest(master)</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/445752/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/454745/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/445750/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/468050/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/454747/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/407378/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/454743/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/369971/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/468047/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/445753/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/310117/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/445751/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/407377/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/468046/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/445755/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/468048/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/454746/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-17.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/445754/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/369972/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/454744/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/468049/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8</a></li><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/369973/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable</a></li></ul></details></details><h5>For more details click <a href="https://test-1-23.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6881/jenkins-report.html">here</a>.</h5></details></html>

jbalsas commented:
ci:forward

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

The pull request will automatically be forwarded to the user `brianchandotcom` if the following test suites pass:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

liferay-continuous-integration commented:
Skipping previously passed test suites:
`ci:test:relevant`
`ci:test:sf`

liferay-continuous-integration commented:
All required test suite(s) passed.
Forwarding pullrequest to `brianchandotcom`.

liferay-continuous-integration commented:
Pull request has been successfully forwarded to https://github.com/brianchandotcom/liferay-portal/pull/95597

liferay-continuous-integration commented:
<strong>Jenkins Build:</strong><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest(master)/6881/">test-portal-acceptance-pullrequest(master)#6881</a><br /><strong>Jenkins Report:</strong><a href="https://test-1-23.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6881/jenkins-report.html">jenkins-report.html</a><br /><strong>Jenkins Suite:</strong> relevant<br /><strong>Pull Request:</strong><a href="https://github.com/liferay-frontend/liferay-portal/pull/464">liferay-frontend#464</a><br /><strong>Spira Release:</strong><a href="https://liferay.spiraservice.net/16/Release/1312.aspx">/Liferay DXP 7.3/Pull Request/ci:test:relevant</a><br /><strong>Spira Release Build:</strong><a href="https://liferay.spiraservice.net/16/Build/56949.aspx">master - kresimir-coko > liferay-frontend - PR#464 - 2020-11-03[04:49:42]</a><br /><strong>Spira Jenkins Build:</strong><a href="https://test-1-8.liferay.com/job/publish-spira-report/7498/">publish-spira-report#7498</a><br />
